test/e2e/clusterclass_rollout.go

```diff
@@ -407,7 +407,7 @@ func assertControlPlane(g Gomega, clusterClassObjects clusterClassObjects, clust
 			ccControlPlaneTemplateMachineTemplateMetadata.Labels,
 		),
 	))
-	g.Expect(controlPlaneMachineTemplateMetadata.Annotations).To(BeEquivalentTo(
+	g.Expect(union(controlPlaneMachineTemplateMetadata.Annotations)).To(BeEquivalentTo(
```

Comment:
What is the pattern behind adding this to in some places but not in others?

Reply:
Hey @sbueringer ,  I only added union() for the cases which are more possible to compare a `nil` and `empty` map,  when only have union() in one of `Expect` or `BeEquivalentTo` as below
```
		g.Expect(machineDeployment.Spec.Template.Annotations).To(BeEquivalentTo(
			union(
				mdTopology.Metadata.Annotations,
				mdClass.Template.Metadata.Annotations,
			),
		))
```
I think other cases are less likely to run into this situation, such as 
```
		g.Expect(infrastructureMachineTemplate.GetAnnotations()).To(BeEquivalentTo(
			union(
				map[string]string{
					clusterv1.TemplateClonedFromGroupKindAnnotation: groupKind(mdClass.Template.Infrastructure.Ref),
					clusterv1.TemplateClonedFromNameAnnotation:      mdClass.Template.Infrastructure.Ref.Name,
				},
				ccInfrastructureMachineTemplate.GetAnnotations(),   ===> We defined expected annotations, and  if infrastructureMachineTemplate.GetAnnotations() are nil or empty, that's a failure we should catch.
			),
		))
		// MachineDeployment InfrastructureMachineTemplate.spec.template.metadata
		g.Expect(infrastructureMachineTemplateTemplateMetadata.Labels).To(BeEquivalentTo(
			ccInfrastructureMachineTemplateTemplateMetadata.Labels,  ===> Usually they either have labels defined or are nil
		))
```

Reply:
Look great, feel free to take it over :-)

Reply:
Opened a PR: https://github.com/kubernetes-sigs/cluster-api/pull/10851